src/full/Agda/Compiler/Common.hs

```diff
@@ -83,18 +83,15 @@ curIF = do
     Nothing -> __IMPOSSIBLE__
     Just mi -> return $ miInterface mi
 
-curSig :: ReadTCState m => m Signature
-curSig = iSignature <$> curIF
-
 curMName :: ReadTCState m => m ModuleName
 curMName = do
   mName <- useTC stCurrentModule
   case mName of
     Nothing -> __IMPOSSIBLE__
     Just name -> return name
 
-curDefs :: TCM Definitions
-curDefs = fmap (HMap.filter (not . defNoCompilation)) $ (^. sigDefinitions) <$> curSig
+curDefs :: ReadTCState m => m Definitions
+curDefs = (HMap.filter (not . defNoCompilation) <$> (^. sigDefinitions)) . iSignature <$> curIF
```

Comment:
I usually reserve `<$>` for `TCM` (the computation monad, resp.), and then use `fmap` for collections like `Maybe`, `Set`, `Map` etc.  I find it very confusing if `<$>` is used for two different monads in one line.
Somehow my tired head spins when looking at this code (not understanding why fmap-fusion cannot be applied).

Reply:
Interesting. Once I got comfortable with `<$>` I found it reads more easily to me, especially in chained expressions. Sort of a mapped version of `$`. With prefix `fmap` I tend to get worn out determining the span of the first argument and end up having to count parenthesis. It's entirely possible I've overused it in a few places like this, though.

Though if `<$>` is confusing for two monads in one line, wouldn't a mix of `fmap` and `<$>` also be confusing? Since `(<$>)` ≡ `fmap`?

(In any case: I've updated this to use `fmap` + `<$>` as it was before inlining)

Reply:
I sometimes use `fmap` infix if I want some of the benefits of `<$>`.

Reply:
> (not understanding why fmap-fusion cannot be applied).

(With a fresher brain:) So why not simply
```haskell
curDefs = HMap.filter (not . defNoCompilation) . (^. sigDefinitions) . iSignature <$> curIF
```
using a single `fmap` rather than two.  I tested it and it type-checks.

Reply:
Thank you, that is much cleaner. I will fixup just that commit and merge once confirming a positive build.

Reply:
[The diff from that rebase](https://github.com/agda/agda/compare/9b9d9b3efa7ffb774321c7f3b46b16e453aa5115..a19270b392dc9a1538c724e38f7643cb395d6934)